### PR TITLE
Fixed errors in soiling.hsu function and tests

### LINKED ISSUE #970: pvlib.soiling.hsu model implementation errors
**Describe the bug**
I ran an example run using the Matlab version of the HSU soiling function and found that the python version did not give anywhere near the same results.  The Matlab results matched the results in the original JPV paper.  As a result of this test, I found two errors in the python implementation, which are listed below:

1.  depo_veloc = {'2_5': 0.004, '10': 0.0009} has the wrong default values.  They are reversed.
The proper dictionary should be: {'2_5': 0.0009, '10': 0.004}.  This is confirmed in the JPV paper and the Matlab version of the function.

2. The horiz_mass_rate is in g/(m^2*hr) but should be in g/(m^2*s).  The line needs to be multiplied by 60x60 or 3600.
The proper line of code should be: 
horiz_mass_rate = (pm2_5 * depo_veloc['2_5']+ np.maximum(pm10 - pm2_5, 0.) * depo_veloc['10'])*3600

When I made these changes I was able to match the validation dataset from the JPV paper, as shown below.
image

### PR DESCRIPTION
**Pull request by jsstein**
- [x] Closes #970
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [n/a] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by wholmgren** [comment]
`E203` can be ignored in the test suite as far as I'm concerned.

**Comment by jsstein** [comment]
I think I have addressed most of the the stickler-ci details (except for the E203 ones which add readability).  I pushed the changes.

File: pvlib/tests/test_soiling.py [hunk 1]
```diff
@@ -119,8 +110,8 @@ def test_hsu_defaults(rainfall_input, expected_output_1):
     accumulation period.
     """
     result = hsu(
-        rainfall=rainfall_input, cleaning_threshold=0.5, tilt=0.0, pm2_5=1.0,
-        pm10=2.0)
+        rainfall=rainfall_input, cleaning_threshold=0.5, tilt=0.0, i
```
Comment (cwhanse) [inline pvlib/tests/test_soiling.py:113]: stray `i` here is causing the test failures

**Comment by jsstein** [comment]
Heavy fingers in vi!

**Comment by cwhanse** [comment]
Thanks @jsstein

File: docs/sphinx/source/whatsnew/v0.8.0.rst [hunk 2]
```diff
@@ -11,6 +11,7 @@ Enhancements
 
 Bug fixes
 ~~~~~~~~~
+* Fixed unit and default value errors in :py:func:`pvlib.soiling.hsu`. (:pull:`XXX`)
```
Comment (kanderso-nrel) [inline docs/sphinx/source/whatsnew/v0.8.0.rst:14]: Missed the PR number in the whatsnew entry

File: pvlib/soiling.py [hunk 3]
```diff
@@ -69,15 +69,15 @@ def hsu(rainfall, cleaning_threshold, tilt, pm2_5, pm10,
...
+        depo_veloc = {'2_5': 0.0009, '10': 0.004}
 
     # accumulate rainfall into periods for comparison with threshold
     accum_rain = rainfall.rolling(rain_accum_period, closed='right').sum()
     # cleaning is True for intervals with rainfall greater than threshold
     cleaning_times = accum_rain.index[accum_rain >= cleaning_threshold]
 
     horiz_mass_rate = pm2_5 * depo_veloc['2_5']\
-        + np.maximum(pm10 - pm2_5, 0.) * depo_veloc['10']
+        + np.maximum(pm10 - pm2_5, 0.) * depo_veloc['10'] * 3600
```
Comment (kanderso-nrel) [inline pvlib/soiling.py:80]: Sorry if I'm thinking about this wrong, but I want to raise two points about this calculation:
- Is it correct to hardcode the 3600?  Shouldn't it be based on the datetime index's interval?  I'm not seeing any documentation requiring the input timeseries to be hourly. 
- Regardless of whatever value that timescale coefficient takes, shouldn't it apply to both particulate sizes?  Right now it only affects pm10.

Comment (jsstein) [reply pvlib/soiling.py:80]: @kanderson-nrel: I agree that I should take another look at the hardcoding of 3600.  About your second point, let me look more carefully at the Matlab version and see how it deals with it.

File: docs/sphinx/source/whatsnew/v0.8.0.rst (hunk 2 above)
Comment (cwhanse) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:14]: I pulled the trigger early

Comment (jsstein) [reply docs/sphinx/source/whatsnew/v0.8.0.rst:14]: I just submitted PR980 that addresses @kanderso-nrel 's comment about arbitrary time intervals.  Next I will submit a demonstration of the hsu function showing that it matches one of the figures in the original JPV paper.

**Pull request merged**